src/main/webapp/app/exam/manage/student-exams/student-exam-timeline/student-exam-timeline.component.ts

```diff
@@ -0,0 +1,373 @@
+import { AfterViewInit, ChangeDetectorRef, Component, OnInit, QueryList, ViewChild, ViewChildren } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { StudentExamService } from 'app/exam/manage/student-exams/student-exam.service';
+import { StudentExam } from 'app/entities/student-exam.model';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { ExamPage } from 'app/entities/exam-page.model';
+import { ExamSubmissionComponent } from 'app/exam/participate/exercises/exam-submission.component';
+import { ExamNavigationBarComponent } from 'app/exam/participate/exam-navigation-bar/exam-navigation-bar.component';
+import { SubmissionService } from 'app/exercises/shared/submission/submission.service';
+import dayjs from 'dayjs/esm';
+import { SubmissionVersion } from 'app/entities/submission-version.model';
+import { Observable, Subscription, map, merge, mergeMap, toArray } from 'rxjs';
+import { ProgrammingSubmission } from 'app/entities/programming-submission.model';
+import { Submission } from 'app/entities/submission.model';
+import { ArtemisDatePipe } from 'app/shared/pipes/artemis-date.pipe';
+import { ChangeContext, Options, SliderComponent } from 'ngx-slider-v2';
+import { FileUploadSubmission } from 'app/entities/file-upload-submission.model';
+import { ProgrammingExamSubmissionComponent } from 'app/exam/participate/exercises/programming/programming-exam-submission.component';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { FileUploadExamSubmissionComponent } from 'app/exam/participate/exercises/file-upload/file-upload-exam-submission.component';
+
+@Component({
+    selector: 'jhi-student-exam-timeline',
+    templateUrl: './student-exam-timeline.component.html',
+    styleUrls: ['./student-exam-timeline.component.scss'],
+})
+export class StudentExamTimelineComponent implements OnInit, AfterViewInit {
+    readonly TEXT = ExerciseType.TEXT;
+    readonly QUIZ = ExerciseType.QUIZ;
+    readonly MODELING = ExerciseType.MODELING;
+    readonly PROGRAMMING = ExerciseType.PROGRAMMING;
+    readonly FILEUPLOAD = ExerciseType.FILE_UPLOAD;
```

Comment:
Personal preference from me:
Only have `readonly ExerciseType = ExerciseType` to reduce the amount of variables